catalyst_voices/apps/voices/lib/pages/overall_spaces/spaces_overview_list_view.dart

```diff
@@ -31,23 +34,62 @@ class _SpacesListViewState extends State<SpacesListView> {
       controller: _scrollController,
       alwaysVisible: true,
       padding: const EdgeInsets.symmetric(horizontal: 6),
-      child: ListView.separated(
-        controller: _scrollController,
-        scrollDirection: Axis.horizontal,
-        padding: const EdgeInsets.only(right: 16, bottom: 24),
-        itemBuilder: (context, index) {
-          final space = Space.values[index];
-          return switch (space) {
-            Space.discovery => DiscoveryOverview(key: ObjectKey(space)),
-            Space.workspace => WorkspaceOverview(key: ObjectKey(space)),
-            Space.voting => VotingOverview(key: ObjectKey(space)),
-            Space.fundedProjects =>
-              FundedProjectsOverview(key: ObjectKey(space)),
-            Space.treasury => TreasuryOverview(key: ObjectKey(space)),
-          };
+      child: BlocSelector<SessionCubit, SessionState, Account?>(
+        selector: (state) {
+          if (state is ActiveAccountSessionState) {
+            return state.account;
+          }
+          return null;
         },
-        separatorBuilder: (context, index) => const SizedBox(width: 16),
-        itemCount: Space.values.length,
+        builder: (context, state) {
+          final spaces = AccessControlUtil.overallSpaces(state);
```

Comment:
how about moving it to `SessionState` so logic related to access lives there?

Widgets should only know how to display stuff